pkg/cli/mirror/catalog_images.go

```diff
@@ -408,56 +366,20 @@ func extractOPMAndCache(ctx context.Context, srcRef image.TypedImageReference, c
 		return err
 	}
 	// cachePath exists, opm binary will be needed to regenerate it
-	tr := tar.NewReader(mutate.Extract(img))
-	opmBinaryExtracted := ""
-	for {
-		header, err := tr.Next()
-
-		// break the infinite loop when EOF
-		if errors.Is(err, io.EOF) {
-			break
-		}
-
-		// skip the file if it is a directory or if file name does not end with `opm`
-		if !strings.HasSuffix(header.Name, opmBinarySuffix) || header.FileInfo().IsDir() {
-			continue
-		}
-
-		var buf bytes.Buffer
-		_, err = buf.ReadFrom(tr)
-		if err != nil {
-			return err
-		}
-
-		targetFileName := filepath.Join(ctlgSrcDir, config.OpmBinDir, header.Name)
-		bytes := buf.Bytes()
-
-		baseDir := filepath.Dir(targetFileName)
-		err = os.MkdirAll(baseDir, 0755)
-		if err != nil {
-			return err
-		}
-
-		f, err := os.Create(targetFileName)
-		if err == nil {
-			defer f.Close()
-		} else {
-			return err
-		}
-
-		_, err = f.Write(bytes)
-		if err != nil {
-			return err
-		}
-		opmBinaryExtracted = targetFileName
-
-		// check for the extracted opm file (it should exist if we found something)
-		_, err = os.Stat(opmBinaryExtracted)
-		if errors.Is(err, os.ErrNotExist) {
-			return fmt.Errorf("opm binary not found after extracting %q from catalog image %v", opmBinaryExtracted, srcRef)
-		}
+	opmBinaryFileName := filepath.Join(ctlgSrcDir, config.OpmBinDir, "opm")
+	err = copyOPMBinary(img, opmBinaryFileName)
```

Comment:
what about the windows-amd64-opm and darwin-amd64-opm binaries (is it ok to ignore them ?)

Reply:
Hi @lmzuccarelli 
`opmBinaryFileName` is the target file where we will store the opm binary corresponding to the runtime platform.
`copyOPMBinary` and more exactly `extractOPMBinary` will take care of extracting the right binary by looking at the entrypoint of the image in `img` parameter. 
That source binary might be `windows-amd64-opm` or `darwin-amd64-opm`, but we're copying it as `opm` so that we simplify the following steps during `processCatalogRefs` as to call command opm under the catalogsource folder.
What do you think?

Reply:
It just occurred to me that entrypoint is not going to work... here's why:

```
$ skopeo --override-os linux inspect docker://registry.redhat.io/openshift4/ose-operator-registry:v4.13 --config | jq '[.config.Entrypoint, .config.Cmd]'
[
  [
    "/bin/registry-server"
  ],
  [
    "--database",
    "/bundles.db"
  ]
]
```

The entrypoint has nothing to do with the opm executable.

Reply:
This is doesn't look like an FBC catalog. Is it?
If that is true, the fact that we only do all this logic when we detect --cache-dir in the CMD will allow us to skip all this for such catalogs.

Reply:
> This was going to be my comment as well, this seems like we lost useful functionality.
Can you explain why you think that?
In the previous implementation we were trying to blindly find anything *opm. In this implementation we are trying to extract the exact binary with which this image is running.